miniscope_io/models/mixins.py

```diff
@@ -23,3 +43,273 @@ def from_yaml(cls: Type[T], file_path: Union[str, Path]) -> T:
         with open(file_path) as file:
             config_data = yaml.safe_load(file)
         return cls(**config_data)
+
+    def to_yaml(self, path: Optional[Path] = None, **kwargs: Any) -> str:
+        """
+        Dump the contents of this class to a yaml file, returning the
+        contents of the dumped string
+        """
+        data_str = self.to_yamls(**kwargs)
+        if path:
+            with open(path, "w") as file:
+                file.write(data_str)
+
+        return data_str
+
+    def to_yamls(self, **kwargs: Any) -> str:
+        """
+        Dump the contents of this class to a yaml string
+
+        Args:
+            **kwargs: passed to :meth:`.BaseModel.model_dump`
+        """
+        data = self._dump_data(**kwargs)
+        return yaml.dump(data, Dumper=YamlDumper, sort_keys=False)
+
+    def _dump_data(self, **kwargs: Any) -> dict:
+        data = self.model_dump(**kwargs) if isinstance(self, BaseModel) else self.__dict__
+        return data
+
+
+class ConfigYAMLMixin(BaseModel, YAMLMixin):
+    """
+    Yaml Mixin class that always puts a header consisting of
+
+     * `id` - unique identifier for this config
+     * `mio_model` - fully-qualified module path to model class
+     * `mio_version` - version of miniscope-io when this model was created
```

Comment:
I like having the model here, and even if this is manually filled, having validation for this could be useful. I've written about the `id` in other parts, so I will skip it here.

However, I'm unsure about the rest because I assume these YAML config files will usually be created by copy-paste and manually filling them out. Having this `mio_version` seems to make these primarily made via a script or CLI command, which I feel will raise the bar without much benefit and cause confusion when hand-edited (which I assume will be the norm).

I feel it'll be more intuitive if we assume the configs will be manually edited. And maybe write a CI flow that documents the `mio_version` that the config YAML was last changed from the commit ID if we need that.

Reply:
Maybe just something like this in an organized way?
https://miniscope-io--84.org.readthedocs.build/en/84/history/config.html

Reply:
> copy-paste and manually filling them out.

I think for most deployments we should assume that people won't be looking at the source code - we assume `pip install miniscope-io` is the default way to get the package. So then they won't have anything to copy/paste, and we need to provide a cli or programmatic interface for dealing with configs. that's sorta the whole idea with the design of having an id field and the way it was implemented as being something that gets autofilled when loaded and then put at the top of the yaml file so we could use a cheap `yaml_peek` method to look through many files quickly. 

So i think this is a pretty easy way to start using the program

```shell
$ pip install miniscope-io
$ mio capture --config wireless
```

and then when i want to make my own config

```shell
$ mio config create wireless --to wireless-custom
# or 
$ mio config create wireless --to my_config_dir/wireless_custom.yaml
```

and then manually edit it.

> CI flow that documents the mio_version that the config YAML was last changed from the commit ID if we need that.

we're looking to support user-custom configs here, so we assume they aren't committing anything to the repository (and don't have to fork it to use it) - versions need to be totally independent from the git flow